airflow/metrics/otel_logger.py

```diff
@@ -0,0 +1,234 @@
+        return Timer()
+
+
+class BaseInstrument(Instrument):
+    """Instrument clss is abstract and must be implemented."""
+
+    def __init__(self, name: str, unit: str = "", description: str = "", attributes: Attributes = None):
+        self.name: str = name
+        self.unit: str = unit
+        self.description: str = description
+        self.attributes: Attributes = attributes
+
+
+class MetricsMap:
+    """Stores Otel Counters."""
+
+    def __init__(self, meter):
+        self.meter = meter
+        self.map = {}
+
+    def clear(self) -> None:
+        self.map.clear()
+
+    def _create_counter(self, name):
+        """Creates a new counter or up_down_counter for the provided name."""
+        otel_safe_name = name[:OTEL_NAME_MAX_LENGTH]
+        if name != otel_safe_name:
+            warnings.warn(
+                f"Metric name `{name}` exceeds OpenTelemetry's name length limit of "
+                f"{OTEL_NAME_MAX_LENGTH} characters and will be truncated to `{otel_safe_name}`."
+            )
+
+        if _is_up_down_counter(name):
+            counter = self.meter.create_up_down_counter(name=otel_safe_name)
+        else:
+            counter = self.meter.create_counter(name=otel_safe_name)
+
+        print(f"--> created {otel_safe_name} as type: {str(type(counter)).split('.')[-1][:-2]}")
+        return counter
+
+    def get_counter(self, name: str, attributes: Attributes = None):
+        """
+        Returns the value of the counter; creates a new one if it does not exist.
+
+        :param name: The name of the counter to fetch or create.
+        :param attributes:  Counter attributes, used to generate a unique key to store the counter.
+        """
+        key: str = name + str(attributes)
+        if key in self.map.keys():
+            return self.map[key]
+        else:
+            new_counter = self._create_counter(name)
+            self.map[key] = new_counter
+            return new_counter
+
+    def del_counter(self, name: str, attributes: Attributes = None) -> None:
+        """
+        Deletes a counter.
+
+        :param name: The name of the counter to fetch or create.
+        :param attributes: Counter attributes which were used to generate a unique key to store the counter.
+        """
+        key: str = name + str(attributes)
+        if key in self.map.keys():
+            del self.map[key]
+
+
+def get_otel_logger(cls) -> SafeOtelLogger:
+    """Get Otel logger"""
+    host = conf.get("metrics", "otel_host")  # ex: "breeze-otel-collector"
+    port = conf.getint("metrics", "otel_port")  # ex: 4318
+    prefix = conf.get("metrics", "otel_prefix")  # ex: "airflow"
+    interval = conf.getint("metrics", "otel_interval_milliseconds")  # ex: 30000
+
+    allow_list = conf.get("metrics", "metrics_allow_list", fallback=None)
+    allow_list_validator = AllowListValidator(allow_list)
+
+    resource = Resource(attributes={SERVICE_NAME: "Airflow"})
+    # TODO:  figure out https instead of http ??
```

Comment:
This doesn't appear to be a simple task.  I've been treating it as a "nice to have" just on principle, but can I bump up the urgency on that if we think it's critical.

Reply:
I would think most enterprise customers would want this. I don't think it blocks this PR, but we should follow soon?

Reply:
Yeah, I agree. Initial research seems to indicate it's not an option, but I find that very hard to believe. I definitely want to look into that more in the near future. As you say, I'm sure it will be a Big Deal for some users.

For posterity, the issue is configuring otel-collector to listen on an `https` connection.